cmd/kubectl-accurate/sub/template_list.go

```diff
@@ -55,49 +56,78 @@ func (o *templateListOpts) Fill(streams genericiooptions.IOStreams, config *gene
 }
 
 func (o *templateListOpts) Run(ctx context.Context) error {
-	if o.template != "" {
-		return o.showNS(ctx, o.template, 0)
+	allNamespaces := &corev1.NamespaceList{}
+	if err := o.client.List(ctx, allNamespaces); err != nil {
+		return fmt.Errorf("failed to list all namespaces: %w", err)
 	}
 
-	templates := &corev1.NamespaceList{}
-	if err := o.client.List(ctx, templates, client.MatchingLabels{constants.LabelType: constants.NSTypeTemplate}); err != nil {
-		return fmt.Errorf("failed to list the template namespaces: %w", err)
-	}
+	sort.Slice(allNamespaces.Items, func(i, j int) bool {
+		return allNamespaces.Items[i].Name < allNamespaces.Items[j].Name
+	})
 
-	for _, ns := range templates.Items {
-		if _, ok := ns.Labels[constants.LabelTemplate]; ok {
-			continue
+	childMap := make(map[string][]*corev1.Namespace)
+	nsMap := make(map[string]*corev1.Namespace)
+
+	for i := range allNamespaces.Items {
+		ns := &allNamespaces.Items[i]
+		nsMap[ns.Name] = ns
+		if parent, ok := ns.Labels[constants.LabelTemplate]; ok {
+			childMap[parent] = append(childMap[parent], ns)
 		}
+	}
 
-		if err := o.showNS(ctx, ns.Name, 0); err != nil {
-			return err
+	for _, children := range childMap {
+		sort.Slice(children, func(i, j int) bool {
```

Comment:
You are sorting too many times.
`allNamespaces.Items` is already sorted, there is no need to sort children here.